cli/azd/test/functional/up_test.go

```diff
@@ -0,0 +1,299 @@
+
+	service, has := showRes.Services["web"]
+	require.True(t, has)
+	require.Equal(t, "dotnet", service.Project.Language)
+	require.Equal(t, "webapp.csproj", filepath.Base(service.Project.Path))
+	require.Equal(t, 1, len(service.Target.ResourceIds))
+
+	_, err = cli.RunCommand(ctx, "deploy")
+	require.NoError(t, err)
+
+	env, err := godotenv.Read(filepath.Join(dir, azdcontext.EnvironmentDirectoryName, envName, ".env"))
+	require.NoError(t, err)
+
+	url, has := env["WEBSITE_URL"]
+	require.True(t, has, "WEBSITE_URL should be in environment after infra create")
+
+	err = probeServiceHealth(t, ctx, url, expectedTestAppResponse)
+	require.NoError(t, err)
+
+	commandRunner := exec.NewCommandRunner(os.Stdin, os.Stdout, os.Stderr)
+	runArgs := newRunArgs("dotnet", "user-secrets", "list", "--project", filepath.Join(dir, "/src/dotnet/webapp.csproj"))
+	secrets, err := commandRunner.Run(ctx, runArgs)
+	require.NoError(t, err)
+
+	contain := strings.Contains(secrets.Stdout, fmt.Sprintf("WEBSITE_URL = %s", url))
+	require.True(t, contain)
+
+	// Ensure `env refresh` works by removing an output parameter from the .env file and ensure that `env refresh`
+	// brings it back.
+	delete(env, "WEBSITE_URL")
+	err = godotenv.Write(env, filepath.Join(dir, azdcontext.EnvironmentDirectoryName, envName, ".env"))
+	require.NoError(t, err)
+
+	//clear dotnet secrets to test if dotnet secrets works when running env refresh
+	runArgs = newRunArgs("dotnet", "user-secrets", "clear", "--project", filepath.Join(dir, "/src/dotnet/webapp.csproj"))
+	secrets, err = commandRunner.Run(ctx, runArgs)
+	require.NoError(t, err)
+
+	_, err = cli.RunCommand(ctx, "env", "refresh")
+	require.NoError(t, err)
+
+	env, err = godotenv.Read(filepath.Join(dir, azdcontext.EnvironmentDirectoryName, envName, ".env"))
+	require.NoError(t, err)
+
+	_, has = env["WEBSITE_URL"]
+	require.True(t, has, "WEBSITE_URL should be in environment after refresh")
+
+	runArgs = newRunArgs("dotnet", "user-secrets", "list", "--project", filepath.Join(dir, "/src/dotnet/webapp.csproj"))
+	secrets, err = commandRunner.Run(ctx, runArgs)
+	require.NoError(t, err)
+
+	contain = strings.Contains(secrets.Stdout, fmt.Sprintf("WEBSITE_URL = %s", url))
+	require.True(t, contain)
+
+	_, err = cli.RunCommand(ctx, "infra", "delete", "--force", "--purge")
+	require.NoError(t, err)
+
+	t.Logf("Running show (again)\n")
+	result, err = cli.RunCommand(ctx, "show", "-o", "json", "--cwd", dir)
+	require.NoError(t, err)
+
+	err = json.Unmarshal([]byte(result.Stdout), &showRes)
+	require.NoError(t, err)
+
+	// Project information should be present, but since we have run infra delete, there shouldn't
+	// be any resource ids.
+	service, has = showRes.Services["web"]
+	require.True(t, has)
+	require.Equal(t, "dotnet", service.Project.Language)
+	require.Equal(t, "webapp.csproj", filepath.Base(service.Project.Path))
+	require.Nil(t, service.Target.ResourceIds)
+}
+
+func Test_CLI_Up_Down_FuncApp(t *testing.T) {
+	t.Skip("azure-dev/834")
```

Comment:
Same comment as above? Will this be removed after your testing is complete?

Reply:
That's correct. It's existing code being moved out of `cli_test.go`. The deployment is still flaky.